### DISCUSSION TITLE
How to assert p95 and P99 LCP details in artillery playwright engine

### DISCUSSION THREAD

**Discussion by Mallikarjun-Roddannavar** [artilleryio/artillery#1810] [Artillery]
Running tests using latest artillery  playwright engine, need help add assertion p95 and P99 LCP details.

Tried with this in config:
ensure:
    p95: 0

but test is not failing

browser.page.LCP.https://test.com
  min: ......................................................................... 2064.2
  max: ......................................................................... 3084.9
  median: ...................................................................... 2416.8
  p95: ......................................................................... 2780
  p99: ......................................................................... 2780

**Comment by juanig1** [comment]
Hey!
The ensure plugin is documented [here](https://www.artillery.io/docs/guides/guides/test-script-reference#ensure---slo-checks).

Using playwright leads to metrics being reported different than the [example](https://www.artillery.io/docs/guides/guides/test-script-reference#threshold-checks) in the docs, as we are actually measuring vuser durations and not response times. We need to tweak the example a bit.

If we check the report for a test using playwright we get:
```
vusers.session_length:
  p95: ............................................... 295.9
  p99: ............................................... 295.9
  ...
```

We see that the names of the metrics we want to assert would be `vusers.session_length.p95` and `vusers.session_length.p99`. So something like this should work:

```
config:
  plugins:    # enable plugin
    ensure: {}
  ensure:     # create assertions
    thresholds:
      # p99 of vusers session length time must be <250:
      - "vusers.session_length.p99": 250
      # p95 of vuser session length time must be <250:
      - "vusers.session_length.p95": 250
  ...
```

Hope this helps!

**Comment by Mallikarjun-Roddannavar** [comment]
@juanig1 This is worked for me, thanks for the details

**Comment by das-en** [comment]
Hi, I am trying to setup a series of similar checks, specifically checking if the requested target url is returning a 200 status. Any advice?

**Reply by juanig1** [reply]
Hey,
When using Playwright, Artillery only knows if your scenario ended successfully.
You could use Playwright [assertions](https://playwright.dev/docs/test-assertions) in your script for more granular checks
